src/replication/adapter/adapter.go

```diff
@@ -122,7 +123,7 @@ func (v *VTag) GetLabels() []string {
 }
 
 // RegisterFactory registers one adapter factory to the registry
-func RegisterFactory(t model.RegistryType, factory Factory) error {
+func RegisterFactory(t model.RegistryType, factory Factory, adapterInfo *model.AdapterInfo) error {
```

Comment:
How about changing the `Factory` to an interface and adding a method `AdapterInfo()`(or we can call it Style/Pattern) to it?
```
type Factory interface{
Create(*model.Registry)(Adapter, error)
AdapterPattern() *Pattern
}
```